<p>Using a list comprehension is not energy efficient.</p>
<p>Prefer the usage of <code>list.copy()</code> which is more energy friendly.</p>
<h2>Noncompliant Code Example</h2>
<pre>
    list = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9]
    list1 = [item for item in list]
</pre>
<h2>Compliant Solution</h2>
<pre>
    list = [1, 2, 3, 4, 5]
    list1 = list.copy()
</pre>
