<p>Do not use copy function of the copy module for shallow copy
<p>Prefer using array.view(), it will do the same but is twice more energy efficient</p>
</p>
<h2>Noncompliant Code Example</h2>
<pre>
    copy.copy(arr)
</pre>
<h2>Compliant Solution</h2>
<pre>
    arr.view()
</pre>
