<p>SVG images generated by common drawing softwares contains unnecessary data: calc layer, metadata, namespaces and comments.</p>
<h2>Noncompliant Code Example</h2>
<pre>
&lt;!-- Created with Inkscape (http://www.inkscape.org/) --&gt;
&lt;svg
    width="210mm"
    height="297mm"
    viewBox="0 0 210 297"
    version="1.1"
    id="svg5"
    inkscape:version="1.2.2 (b0a8486541, 2022-12-01)"
    sodipodi:docname="dessin2.svg"
    xmlns:inkscape="http://www.inkscape.org/namespaces/inkscape"
    xmlns:sodipodi="http://sodipodi.sourceforge.net/DTD/sodipodi-0.dtd"
    xmlns="http://www.w3.org/2000/svg"
    xmlns:svg="http://www.w3.org/2000/svg"&gt;
    &lt;g
            inkscape:label="Calque 1"
            inkscape:groupmode="layer"
            id="layer1"&gt;
        &lt;circle
            style="fill:#ff00ff;stroke-width:0.264583"
            id="path111"
            cx="104.02724"
            cy="152.19028"
            r="73.177132" /&gt;
    &lt;/g&gt;
&lt;/svg&gt;
</pre>
<h2>Compliant Solution</h2>
<pre>
&lt;svg
    width="210mm"
    height="297mm"
    viewBox="0 0 210 297"
    xmlns="http://www.w3.org/2000/svg"
    xmlns:svg="http://www.w3.org/2000/svg"&gt;
    &lt;g&gt;
        &lt;circle
            style="fill:#ff00ff;stroke-width:0.264583"
            id="path111"
            cx="104.02724"
            cy="152.19028"
            r="73.177132" /&gt;
    &lt;/g&gt;
&lt;/svg&gt;
</pre>
