<p><b>Use generator comprehension instead of list comprehension in for loop declaration.</b></p>
<p>Python generators resemble lazy lists from other programming languages: when iterated over, they compute their values on the fly. They lack some list behaviors (indexing, len method, ...) but are memory-efficient, as they do not store each of their values in memory, unlike lists. Thus, when declared in a for-loop declaration, list comprehensions can be safely replaced with generator comprehensions. </p>
<p>For more details on list comprehensions vs generator comprehensions, see <a href="https://docs.python.org/3/howto/functional.html#generator-expressions-and-list-comprehensions">Python documentation</a>.</p>
<h2>Noncompliant Code Example</h2>
<pre>
for var in [var2 for var2 in range(100)]:
    ...
</pre>
<h2>Compliant Solution</h2>
<pre>
for var in (var2 for var2 in range(100)):
    ...
</pre>
