<p>Using <code>str.format()</code> to perform a string concatenation is not energy efficient.</p>
<p>Prefer using <code>+</code> operator to concatenate String</p>
<h2>Noncompliant Code Example</h2>
<pre>
    '{}{}'.format(string_1, string_2)
</pre>
<h2>Compliant Solution</h2>
<pre>
    string_1+string_2
</pre>
