<p>Use <code>PreparedStatement</code> instead of <code>Statement</code>, because SQL will only commit the query once, whereas if you used only one statement, it would commit the query every time and thus induce unnecessary calculations by the CPU and therefore superfluous energy consumption.</p>
<h2>Noncompliant Code Example</h2>
<pre>
    public void select() {
        Statement statement = connection.createStatement();
        statement.executeUpdate("INSERT INTO persons(id, name) VALUES(2, 'Toto')");  // Noncompliant
    }
</pre>
<h2>Compliant Solution</h2>
<pre>
    public void select() {
        PreparedStatement statement = connection.prepareStatement(INSERT INTO persons(id, name) VALUES(?, ?));

        statement.setInt(1, 2);
        statement.setString(2, "Toto");
        statement.executeQuery();
    }
</pre>
