<p>Database servers have to resolve schema fields when using asterisk symbol (*). Knowing and using the schema saves CPU cycles and network transfer.</p>
<h2>Noncompliant Code Example</h2>
<pre>
    public void foo() {
        ...
        String baseQuery = "SELECT * FROM users"; // Noncompliant

        ...
    }

</pre>
<h2>Compliant Solution</h2>
<pre>

    public void foo() {
        ...
        String query = "SELECT id, name, address FROM users ";
        ...
   }

</pre>
