<p>Inside complex code parts (for exemple multiple loops, complex data constructions...), avoid using try...catch...finally.</p>
<p>When an exception is thrown, a variable (the exception itself) is created in a catch block and it's destruction consumes unnecessary CPU cycles and RAM. Prefer using logical tests in this cases.</p>
<h2>Noncompliant Code Example</h2>
<pre>
try
{
  $picture = PDF_open_image_file($PDF, "jpeg", $imgFile, "", 0); // This is the original statement, this works on PHP4
}
catch(Exception $ex)
{
  $msg = "Error opening $imgFile for Product $row['Identifier']";
  throw new Exception($msg);
}

</pre>
<h2>Compliant Solution</h2>
<pre>
//try
if (file_exists($imgFile)) {
    $picture = PDF_open_image_file($PDF, "jpeg", $imgFile, "", 0);
}

//catch
if (!$picture) {
   $msg = "Error opening $imgFile for Product $row['Identifier']";
   print $msg;
}
</pre>
<h2>The three sources of impacts of a code identified are: </h2>
- Energy (measured in joules)
<br />    - Transfer (measured in Bytes)
<br />    - Storage (measured in Bytes)<br />
<br />The control of these 3 impacts allows to lengthen the life of the terminals as well as reduce their energy consumption.
<br />The ETSdiff tool allows measuring a differential on these three values and in a given context (database and fixed measurement environment).
<br />The results generated by ETSdiff must help define the interest of the rule reported by Sonarqube in the context of the code analyzed.
<br />
<h2>Case for a 1GB database:</h2>
<figure >
    <img src="https://live.staticflickr.com/65535/52622382871_f19da08db4_o.png" alt="ETSdiff percent comparison" />
</figure>
<h2>Total:</h2>
<table border="1" style="border-collapse: collapse; width: 100%; height: 40px;border:none;">
    <tbody style="text-align: center;">
    <tr style="height: 10px;">
        <td style="width: 33.3333%; height: 10px;border:none;"></td>
        <td style="width: 33.3333%; height: 10px;">
            <div>
                <div><strong>Compliant</strong></div>
            </div>
        </td>
        <td style="width: 33.3333%; height: 10px;">
            <div>
                <div><strong>Non-compliant</strong></div>
            </div>
        </td>
    </tr>
    <tr style="height: 10px;">
        <td style="width: 33.3333%; height: 10px;">
            <div>
                <div><strong>Energy</strong></div>
            </div>
        </td>
        <td style="width: 33.3333%; height: 10px;">
            <div>
                <div><span>515.855638</span></div>
            </div>
        </td>
        <td style="width: 33.3333%; height: 10px;">
            <div>
                <div><span>516.9188409999999</span></div>
            </div>
        </td>
    </tr>
    <tr style="height: 10px;">
        <td style="width: 33.3333%; height: 10px;">
            <div>
                <div><strong>Transfer</strong></div>
            </div>
        </td>
        <td style="width: 33.3333%; height: 10px;">
            <div>
                <div><span>1579453</span></div>
            </div>
        </td>
        <td style="width: 33.3333%; height: 10px;">
            <div>
                <div><span>1579457</span></div>
            </div>
        </td>
    </tr>
    <tr style="height: 10px;">
        <td style="width: 33.3333%; height: 10px;">
            <div>
                <div><strong>Storage</strong></div>
            </div>
        </td>
        <td style="width: 33.3333%; height: 10px;">
            <div>
                <div><span>637549804</span></div>
            </div>
        </td>
        <td style="width: 33.3333%; height: 10px;">
            <div>
                <div><span>637549804</span></div>
            </div>
        </td>
    </tr>
    </tbody>
</table>
