<p>Databases servers have to solve fields regarding the schema. Knowing and using the schema save CPU cycles and network transfer.</p>
<h2>Noncompliant Code Example</h2>
<pre>
    public function foo() {
        ...
        $baseQuery = "SELECT * FROM users"; // Noncompliant

        ...
    }
</pre>
<h2>Compliant Solution</h2>
<pre>
    public function foo() {
        ...
        $baseQuery = "SELECT id,name, adress FROM users ";
        ...
   }
</pre>
<h2>The three sources of impacts of a code identified are: </h2>
- Energy (measured in joules)
<br />    - Transfer (measured in Bytes)
<br />    - Storage (measured in Bytes)<br />
<br />The control of these 3 impacts allows to lengthen the life of the terminals as well as reduce their energy consumption.
<br />The ETSdiff tool allows measuring a differential on these three values and in a given context (database and fixed measurement environment).
<br />The results generated by ETSdiff must help define the interest of the rule reported by Sonarqube in the context of the code analyzed.
<br />

<h2>Case for a 1GB database:</h2>
<figure style="text-align: center;">
    <img src="https://live.staticflickr.com/65535/52622636584_52938fcf7e_o.png" alt="ETSdiff percent comparison" />
</figure>

<h2>Total:</h2>
<table border="1" style="border-collapse: collapse; width: 100%; height: 40px;border:none;">
    <tbody style="text-align: center;">
    <tr style="height: 10px;">
        <td style="width: 33.3333%; height: 10px;border:none;"></td>
        <td style="width: 33.3333%; height: 10px;">
            <div>
                <div><strong>Compliant</strong></div>
            </div>
        </td>
        <td style="width: 33.3333%; height: 10px;">
            <div>
                <div><strong>Non-compliant</strong></div>
            </div>
        </td>
    </tr>
    <tr style="height: 10px;">
        <td style="width: 33.3333%; height: 10px;">
            <div>
                <div><strong>Energy</strong></div>
            </div>
        </td>
        <td style="width: 33.3333%; height: 10px;">
            <div>
                <div><span>0.040610499999999994</span></div>
            </div>
        </td>
        <td style="width: 33.3333%; height: 10px;">
            <div>
                <div><span>0.065223</span></div>
            </div>
        </td>
    </tr>
    <tr style="height: 10px;">
        <td style="width: 33.3333%; height: 10px;">
            <div>
                <div><strong>Transfer</strong></div>
            </div>
        </td>
        <td style="width: 33.3333%; height: 10px;">
            <div>
                <div><span>779232</span></div>
            </div>
        </td>
        <td style="width: 33.3333%; height: 10px;">
            <div>
                <div><span>2697937</span></div>
            </div>
        </td>
    </tr>
    <tr style="height: 10px;">
        <td style="width: 33.3333%; height: 10px;">
            <div>
                <div><strong>Storage</strong></div>
            </div>
        </td>
        <td style="width: 33.3333%; height: 10px;">
            <div>
                <div><span>637548827</span></div>
            </div>
        </td>
        <td style="width: 33.3333%; height: 10px;">
            <div>
                <div><span>637548827</span></div>
            </div>
        </td>
    </tr>
    </tbody>
</table>